tools/quickjs-handle.mjs

```diff
@@ -0,0 +1,264 @@
+import { DefaultIntrinsics, getQuickJS } from "quickjs-emscripten";
+
+const globalsCode = `
+    class ProcessError extends Error {
+      constructor(message) {
+        super(message);
+        this.name = "ProcessError";
+      }
+    }
+
+    // add all stuff from https://cookbook_ao.g8way.io/references/ao.html
+    let ao = {
+      _version: "0.0.3",
+      id: "",
+      _module: "",
+      authorities: [],
+      _ref: 0,
+      outbox: {
+        Messages: [],
+        Spawns: []
+      },
+      env: {}
+    }
+    
+    let currentState = {};
+    
+    BigInt.prototype.toJSON = function () {
+      return this.toString();
+    };
+    
+    // to be called by SDK when evaluating for already cached state - same as WASM handlers
+    function __initState(newState) {
+      console.log('__initState', newState);
+      currentState = newState;
+    }
+    
+    // to be called by SDK after evaluating a message - same as WASM handlers
+    function __currentState() {
+      return JSON.stringify(currentState);
+    }
+    
+    function __getOutbox() {
+      return JSON.stringify(ao.outbox);
+    }
+`;
+
+// the example smart contract code loaded from Arweave
+// note: processCode MUST HAVE 'function handle(state, message, aoGlobals)'
+const processCode = `
+  function handle(state, message, aoGlobals) {
```

Comment:
keep in mind that aoGlobal is probably no longer neede - and all the stuff that was inside in aoGlobal should be kept in ao.env